deploy-web/src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -85,29 +93,43 @@ export const KeplrWalletProvider = ({ children }) => {
 
   useEffect(() => {
     if (isWindowLoaded && isSettingsInit) {
-      if (!!window.keplr) {
-        setIsKeplrInstalled(true);
+      if (!!window.keplr || !!window.leap) {
+        if (!!window.keplr) {
+          setIsKeplrInstalled(true);
+        }
+
+        if (!!window.leap) {
+          setIsLeapInstalled(true);
+        }
+
+        if (localStorage.getItem("wallet_autoconnect")) {
+          const storedWallet = localStorage.getItem("wallet_autoconnect");
 
-        window.keplr.defaultOptions = {
-          sign: {
-            preferNoSetMemo: true
+          if (storedWallet === Wallets.KEPLR) {
+            window.wallet = window.keplr;
+          } else if (storedWallet === Wallets.LEAP) {
+            window.wallet = window.leap;
           }
-        };
 
-        if (localStorage.getItem("keplr_autoconnect")) {
+          window.wallet.defaultOptions = {
+            sign: {
+              preferNoSetMemo: true
+            }
+          };
+
           loadWallet();
         } else {
           setIsWalletLoaded(true);
         }
 
-        window.addEventListener("keplr_keystorechange", onKeystoreChange);
+        window.addEventListener("keplr_keystorechange", () => onKeystoreChange(Wallets.KEPLR));
+        window.addEventListener("leap_keystorechange", () => onKeystoreChange(Wallets.LEAP));
```

Comment:
Though, I made the changes, but, Was it not working properly? because it should work, as we can use arrow functions in event listeners. And, since, these are two different events then there should be no issue in removing listeners as well.

Reply:
The events were added correctly, but the `removeEventListener` was not removing them since they were not the same arrow functions as the ones added. This explains the issue: https://www.semicolonandsons.com/code_diary/javascript/anonymous-functions-do-not-work-with-remove-event-listener

Reply:
Oh! Nice. Makes sense, we have not given the same function reference.
Thanks for this.